### ISSUE TITLE
OPAM file generation doesn't respect (lang dune)

### ISSUE THREAD

**Issue by ulrikstrid** [ocaml/dune#3700]
## Expected Behavior
If I set `(lang dune 2.5)` in my dune-project and then put `(dune (>= 1.11))` in `depends` I expect it to warn me about incompatibility or even update the dune version to `2.5`.

## Actual Behavior
It just generates the OPAM file with the dune version i supply which is then invalid.

## Specifications

- Version of `dune` (output of `dune --version`): 2.6.2
- Version of `ocaml` (output of `ocamlc --version`): 4.10.0
- Operating system (distribution and version): macOS Catalina

**Comment by rgrinberg** [comment]
This is unfortunate but intentional. It's a little hard to reverse engineer the user's hand written constraint so we just don't bother.

@kit-ty-kate I suppose we could reverse engineer simple constraints like this one and tell the user when they're obviously wrong. What do you think?

**Comment by kit-ty-kate** [comment]
yeah sounds relatively easy. A function like this one https://github.com/ocaml/dune/pull/3667/files#diff-2313f7848095c19024e87d91e94fb5f1R135 but looking for "too short constraints" and emitting a warning, should be enough.

**Comment by rgrinberg** [comment]
Sounds good. Do you want to tackle this?

**Comment by kit-ty-kate** [comment]
why not but I'm in vacations right now. So maybe if I get bored. Otherwise afterwards.

**Comment by rgrinberg** [comment]
Alright. Enjoy your vacation then.

@ulrikstrid Would you like to work on this in the near future perhaps?

**Comment by ulrikstrid** [comment]
I might be able to look into it. Any pointers on where to start? Is the PR mentioned above a good reference?

**Comment by rgrinberg** [comment]
Yeah, the PR above should be sufficient. Let us know if you have other questions.
